src/main/java/aharon/sudoku/Sudoku.java

```diff
@@ -0,0 +1,85 @@
+package aharon.sudoku;
+
+import java.util.*;
+
+public class Sudoku
+{
+
+    private int[][] board;
+
+    public Sudoku(int[][] board) {
+        this.board = board;
+    }
+
+    public List<String> getErrors()
+    {
+        List<String> errors = new ArrayList<>();
+
+        for (int col = 0; col < 9; col++) {
+            int[] numsPassed = new int[9];
+            int countPassed = 0;
+
+            for (int row = 0; row < 9; row++) {
+                int currentNumber = board[row][col];
+
+                for (int np = 0; np < countPassed; np++) {
+                    if (numsPassed[np] == currentNumber) {
+                        String error = "Duplicate of " + currentNumber + " in row " + row + ", column " + col;
+                        errors.add(error);
+                    }
+                }
```

Comment:
If you use currentNumber as your index into the array, then you don't need to loop through the whole array

Reply:
Set

numsPassed[currentNumber] = 1

When you encounter the number.

Then you can easily tell if you've seen it without looking through numsPassed